### DISCUSSION TITLE
Implementing an app

### DISCUSSION THREAD

**Discussion by JYANG2002** [nasa/cFS#415] [Q&A]
Hi, I am fairly new to cFS and I have a question on how to properly add something to the app directory. Like say I want to clone a https://github.com/nasa/DS in there. However, when I build it I get a fatal error cfs_utils.h.

**Comment by skliper** [comment]
If you are getting a fatal error my guess would be version compatibility issue.  Older app releases won't work with the main (Draco development cycle) branch in cFS.  CF main does work with both Caelum-rc4 and main, and we are working on releasing 9 other apps that are Caelum/Draco compliant. An example of an older distribution can be found at https://github.com/OpenSatKit/OpenSatKit (cFE 6.7.1 baseline).

**Reply by JYANG2002** [reply]
Some of the other apps work like HK LC and CS. Other apps I'm having trouble with are DS MD MM and FM and most of the errors I'm getting at occurs when it #include "cfs_utils.h" causing it to terminate with error 1 and 2

**Reply by skliper** [reply]
That will all depend on which versions of the framework and apps you are using.
